tests/dbgp/dbgpclient.php

```diff
@@ -96,95 +102,136 @@ function fixFilePath( $m )
 		return " {$m[1]}=\"file://{$fm[1]}\"";
 	}
 
-	function doRead( $conn )
+	function doRead( $conn, string $transaction_id = null )
 	{
-		stream_set_timeout( $conn, 0, 1000 );
+		stream_set_timeout($conn, 1);
 		do {
-			$end = true;
-			do {
-				$header = stream_get_line( $conn, 10240, "\0" );
-			} while ( $header === false );
-			$read   = stream_get_line( $conn, 102400, "\0" );
+			$trans_id = null;
+			$length = 0;
+			while ("\0" !== ($char = fgetc($conn))) {
+				if ($char === false) {
+					echo "read a false for $transaction_id".PHP_EOL;
+					return null;
+				}
+				if (!is_numeric($char)) {
+					echo "read a non-number for $transaction_id".PHP_EOL;
+					return null;
+				}
+				$length = $length * 10 + (int)$char;
+			}
+
+			$read = '';
+			while (0 < $length) {
+				$data = fread($conn, $length);
+				if ($data === false) {
+					echo "read a false for $transaction_id".PHP_EOL;
+					return null;
+				}
+				$length -= strlen($data);
+				$read .= $data;
+			}
+			$char = fgetc($conn);
+			if ($char !== "\0") {
+				echo 'must end with \0' . PHP_EOL;
+			}
 
 			// sanitize
-			$read = preg_replace( '@\s(appid|id)="\d+?"@', ' \\1=""', $read );
-			$read = preg_replace( '@\s(idekey)="[^"]+?"@', ' \\1=""', $read );
-			$read = preg_replace( '@\s(xdebug:language_version)="[^"]+?"@', ' \\1=""', $read );
-			$read = preg_replace( '@(engine\sversion)="[^"]+?"@', '\\1=""', $read );
-			$read = preg_replace( '@(2002-20[0-9]{2})@', '2002-2099', $read );
-			$read = preg_replace_callback( '@\s(fileuri|filename)="file:///(.+?)"@', 'self::fixFilePath', $read );
+			$read = preg_replace('@\s(appid|id)="\d+?"@', ' \\1=""', $read);
```

Comment:
Please don't change away from my coding standard. I know it's not PSR-2, but I also don't want to change it. In general, I wouldn't want any only-white-space-changes mixed in with functional changes.